src/components/Asset/AssetActions/Pool/Remove/index.tsx

```diff
@@ -137,20 +134,18 @@ export default function Remove({
       .dividedBy(100)
       .mul(new Decimal(poolInfoUser.poolShares))
       .toString()
-
-    setAmountPoolShares(`${amountPoolShares.slice(0, 18)}`)
+    setAmountPoolShares(amountPoolShares)
   }
 
-  function handleMaxButton(e: ChangeEvent<HTMLInputElement>) {
+  async function handleMaxButton(e: ChangeEvent<HTMLInputElement>) {
```

Comment:
nothing `async` going on here but guess some state setting works more reliably with it here?